components/common-go/namegen/workspaceid.go

```diff
@@ -36,8 +36,12 @@ func GenerateWorkspaceID() (string, error) {
 	if err != nil {
 		return "", err
 	}
+	t1 := strings.Join([]string{s1, s2}, "-")
+	if len(t1) > 23 {
+		t1 = t1[0:23]
+	}
 
-	return strings.Join([]string{s1, s2, s3}, "-"), nil
+	return strings.Join([]string{t1, s3}, "-"), nil
```

Comment:
Could you please add a test for this case? Such that we don't repeat this error next time we need to make adjustments to the pattern.